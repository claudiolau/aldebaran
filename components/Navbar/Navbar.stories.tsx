import React from "react";
import { Navbar } from "./Navbar";
import { ComponentMeta } from "@storybook/react";
import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  Stories,
  PRIMARY_STORY,
} from "@storybook/addon-docs";

export default {
  /* ðŸ‘‡ The title prop is optional.
   * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: "Navbar",
  component: Navbar,
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle>
            A basic navigation bar geared for personal websites.
          </Subtitle>
          <Description />
          <Primary />
          <ArgsTable story={PRIMARY_STORY} />
          <Stories />
        </>
      ),
    },
  },
} as ComponentMeta<typeof Navbar>;

export const Basic = () => (
  <Navbar navtitle={["Home", "Github", "Blog", "Gear"]} />
);
