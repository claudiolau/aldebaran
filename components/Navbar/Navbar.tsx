import * as React from "react";
import styles from "./_navbar.module.scss";
import { BrowserRouter as Router, NavLink } from "react-router-dom";

type NavbarProps = {
  navtitle: string[];
};

export const Navbar: React.FC<NavbarProps> = ({ navtitle }) => {
  return (
    <Router>
      <ul className={styles.navbar}>
        {navtitle.map((title, index) => {
          return (
            <>
              <li key={`${index}`}>
                <NavLink
                  key={`${index}`}
                  to={`/${title}`}
                  className={({ isActive }) =>
                    isActive ? styles.active : styles.inactive
                  }
                >
                  {title}
                </NavLink>
              </li>
            </>
          );
        })}
      </ul>
    </Router>
  );
};
