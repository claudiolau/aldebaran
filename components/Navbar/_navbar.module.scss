.navbar {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1rem;
  color: #fff;
}

.navbar li {
  padding: 1rem;

  a:hover {
    box-shadow: inset 100px 0 0 0 rgba(0, 0, 0, 0.05);
    border-radius: 5px;
  }
  > .active {
    box-shadow: inset 100px 0 0 0 rgba(0, 0, 0, 0.05);
    border-radius: 5px;
  }
  > .inactive {
    box-shadow: none;
  }
}
