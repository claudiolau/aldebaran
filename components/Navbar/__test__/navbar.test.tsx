import * as React from "react";
import { render } from "@testing-library/react";
import { Navbar } from "../Navbar";

test("items navbar", async () => {
  const { asFragment } = render(<Navbar navtitle={["test"]} />);
  expect(asFragment()).toMatchSnapshot();
});
