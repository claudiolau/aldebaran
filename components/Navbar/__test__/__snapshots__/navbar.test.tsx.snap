// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`items navbar 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/test"
      >
        test
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`loads items eventually 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/test"
      >
        test
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`loads items navbar 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/test"
      >
        test
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
